### PR TITLE
Fixing file leaks in tests (preadapting to proposed `TemporaryDirectoryAllocator.dispose` change)

### PR DESCRIPTION
**Pull request by jglick**
Fixing some issues in tests that could result in file handles being left open, especially on Windows.

### Testing done

Ran tests against https://github.com/jenkinsci/jenkins-test-harness/pull/198 which is stricter about cleanup.

### Proposed changelog entries

- N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8006"></a>

### CODE REVIEW

**Comment by jglick** [comment]
```
[ERROR] hudson.LauncherTest.remotable  Time elapsed: 4.281 s  <<< ERROR!
java.nio.file.FileSystemException: C:\Jenkins\workspace\Core_jenkins_PR-8006\test\target\j h12573638483276859145\log: The process cannot access the file because it is being used by another process
	at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:92)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
	at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:275)
	at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
	at java.base/java.nio.file.Files.deleteIfExists(Files.java:1191)
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.delete(TemporaryDirectoryAllocator.java:135)
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.delete(TemporaryDirectoryAllocator.java:130)
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.dispose(TemporaryDirectoryAllocator.java:98)
	at org.jvnet.hudson.test.TestEnvironment.dispose(TestEnvironment.java:84)
	at …
```

smells like a genuine test bug.

**Comment by jglick** [comment]
https://github.com/jenkinsci/jenkins/pull/8006/checks?check_run_id=13524678256 also probably a test bug; does not wait for all builds it triggered to complete.

File: test/src/test/java/hudson/LauncherTest.java
Comment (jglick) [inline test/src/test/java/hudson/LauncherTest.java:194]: (hide whitespace to review)

File: test/src/test/java/hudson/model/DirectoryBrowserSupportTest.java
Comment (jtnord) [inline test/src/test/java/hudson/model/DirectoryBrowserSupportTest.java:923]: why this cleanup?  isn't this exactly what the test harness does (and if that can not clean up because the file is locked why would `rmdir` be able to?

**Review by jtnord — COMMENTED** [review]
one call to rmdir looks strange.

I'm guessing you have a reason for it - in which case it could do with an inline comment explaining why, because otherwise it looks like it would just do the same as a cleanup in `JenkinsRule`

File: test/src/test/java/hudson/model/DirectoryBrowserSupportTest.java
Comment (jglick) [reply test/src/test/java/hudson/model/DirectoryBrowserSupportTest.java:923]: https://github.com/jenkinsci/jenkins-test-harness/pull/198#discussion_r501366464

**Review by jtnord — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Comment by basil** [comment]
`LauncherTest` still flaky on Windows: https://github.com/jenkinsci/jenkins/pull/8728/checks?check_run_id=19150437978

**Comment by jglick** [comment]
Thanks, will take a look soon.

**Pull request merged**
